cmd/thanos/config.go

```diff
@@ -217,3 +217,15 @@ func (ac *alertMgrConfig) registerFlag(cmd extflag.FlagClause) *alertMgrConfig {
 		StringsVar(&ac.alertExcludeLabels)
 	return ac
 }
+
+type ruleRWConfig struct {
+	remoteWrite bool
+	configPath  *extflag.PathOrContent
+}
+
+func (rc *ruleRWConfig) registerFlag(cmd extflag.FlagClause) *ruleRWConfig {
+	cmd.Flag("remote-write", "If true, directs ruler to remote-write evaluated samples to the server configured by 'remote-write.config'.").
```

Comment:
Why not check if config is provided and only then enable remote write? ;p

Reply:
Important! User will expect remote write can be added to the TSDB ruler also.. We need to make sure that is not the case, and if you provide remote write it will make ruler stateless (except the WAL buffer)